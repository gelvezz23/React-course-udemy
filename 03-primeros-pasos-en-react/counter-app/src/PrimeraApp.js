/** @format */

import React from 'react';
import PropTypes from 'prop-types';

const PrimeraApp = ({ saludo }) => {
	return (
		// este es un fragment abreviado
		<>
			{/* se puede hacer asi pero no se usa <h1>{props.saludo}</h1> */}
			<h1>{saludo}</h1>
			<p>esto es una props con destructuracion</p>
		</>
	);
};

PrimeraApp.propTypes = {
	saludo: PropTypes.string.isRequired,
};

export default PrimeraApp;
