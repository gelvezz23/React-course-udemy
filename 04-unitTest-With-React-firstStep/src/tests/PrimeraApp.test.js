/** @format */
import React from 'react';
// import { render } from '@testing-library/react';
import PrimeraApp from '../PrimeraApp';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';

describe('Pruebas en <PrimeraApp/>', () => {
	// test('Debe de mostrar el mensaje hola', () => {
	// 	const saludo = 'hola';
	// 	// es una funcion de jest que recibe un componente
	// 	// getByText es un metodo de render
	// 	const { getByText } = render(<PrimeraApp saludo={saludo} />);

	// 	// sin destructurar wrapper = render(<PrimeraApp saludo={saludo} />);

	// 	//sin usar inzame
	// 	expect(getByText(saludo)).toBeInTheDocument();
	// });

	test('Debe mostrar <PrimeraApp/> correctamente', () => {
		const saludo = 'hola';

		const wrapper = shallow(<PrimeraApp saludo={saludo} />);

		expect(wrapper).toMatchSnapshot();
	});
});
