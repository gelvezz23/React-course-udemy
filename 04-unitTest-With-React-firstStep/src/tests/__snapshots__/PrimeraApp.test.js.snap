// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <PrimeraApp/> Debe mostrar <PrimeraApp/> correctamente 1`] = `
<Fragment>
  <h1>
    hola
  </h1>
  <p>
    Soy un subtitulo por defecto
  </p>
</Fragment>
`;
